src/test/java/uk/gov/justice/log/main/SearchHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class SearchHandlerTest {
+    private final static String RESPONSE_STR_PATH = "src/test/resources/response.json";
+
+    private final String queryStr = "{}{\"query\":{\"bool\":{\"should\":" +
+            "{\"terms\":{\"message\":[\"202\"]}},\"must\":{\"range\":" +
+            "{\"@timestamp\":{\"from\":1486654918164,\"to\":1486658518164}}}}}}";
+    @Rule
+    public ExpectedException expectedExption = ExpectedException.none();
+
+    @Mock
+    PropertyReader propertyReader;
+
+    @Mock
+    SearchService searchService;
+
+    @Mock
+    ElasticSearchQueryBuilder elasticSearchQueryBuilder;
+
+    @Mock
+    Response response;
+
+    @Mock
+    RestClient restClient;
+
+    @Mock
+    RestConfig restConfig;
+
+    @Mock
+    SearchCriteria searchCriteria;
+
+    @Mock
+    RestClientFactory restClientFactory;
+
+    @Mock
+    SearchLogsFactory searchLogsFactory;
+
+    @Mock
+    ResultsPrinterFactory resultsPrinterFactory;
+
+    @Mock
+    ResultsPrinter resultsPrinter;
+
+    @Mock
+    ConnectionManager connectionManager;
+
+    @InjectMocks
+    SearchHandler searchHandler;
+
+
+    @Test
+    public void shouldSearchViaSearchService() throws Exception {
+        final HttpEntity query = new NStringEntity(queryStr);
+        byte[] encoded = Files.readAllBytes(Paths.get(RESPONSE_STR_PATH));
+        final String responseStr = new String(encoded, "UTF-8");
+        final HttpEntity responseEntity = new NStringEntity(responseStr, ContentType.APPLICATION_JSON);
+        final JSONArray array = new JSONArray();
+        when(propertyReader.searchCriteria()).thenReturn(searchCriteria);
+        when(propertyReader.restConfig()).thenReturn(restConfig);
+        when(restClientFactory.restClient()).thenReturn(restClient);
+        when(searchLogsFactory.create(restClient, elasticSearchQueryBuilder)).thenReturn(searchService);
+        when(resultsPrinterFactory.createResultsPrinters("yes")).thenReturn(resultsPrinter);
+        when(searchService.search()).thenReturn(response);
+        when(elasticSearchQueryBuilder.searchCriteria()).thenReturn(searchCriteria);
+        when(elasticSearchQueryBuilder.entityQuery()).thenReturn(query);
+        when(elasticSearchQueryBuilder.query()).thenReturn(queryStr);
+        when(searchCriteria.getFromTime()).thenReturn("fromtime");
+        when(searchCriteria.getToTime()).thenReturn("totime");
+
+        when(response.getEntity()).thenReturn(responseEntity);
+
+        final SearchHandler searchLogs = new SearchHandler(elasticSearchQueryBuilder, restClient, searchLogsFactory, resultsPrinterFactory);
+
+        searchLogs.searchLogs("yes");
+
+        final String responseStrActual = EntityUtils.toString(response.getEntity());
+        final Integer hits = JsonPath.read(responseStrActual, "$.responses[0].hits.total");
+        final JSONArray messages = JsonPath.read(responseStrActual, "$.responses[0].hits..message");
+
+        assertThat(hits, is(1));
```

Comment:
Now separated extraction to a ResultExtractor and created a ResultExtractor test for asserting the extracted response